modules/nw-ipfailover-configuration.adoc

```diff
@@ -0,0 +1,160 @@
+    type: Recreate
+  replicas: 2
+  selector:
+    matchLabels:
+      ipfailover: kuard-keepalived
+  triggers:
+    type: ConfigChange
+  template:
+    metadata:
+      labels:
+        ipfailover: kuard-keepalived
+    spec:
+      serviceAccountName: ipfailover
+      privileged: true
+      hostNetwork: true
+      nodeSelector:
+        node-role.kubernetes.io/worker: ""
+      containers:
+      - name: openshift-ipfailover
+        image: quay.io/openshift/ose-keepalived-ipfailover:latest 
+        ports:
+        - containerPort: 63000
+          hostPort: 63000
+        imagePullPolicy: IfNotPresent
+        securityContext:
+          privileged: true
+        volumeMounts:
+        - name: lib-modules
+          mountPath: /lib/modules
+          readOnly: true
+        - name: host-slash
+          mountPath: /host
+          readOnly: true
+          mountPropagation: HostToContainer
+        - name: etc-sysconfig
+          mountPath: /etc/sysconfig
+          readOnly: true
+        - name: config-volume
+          mountPath: /etc/keepalive
+        env:
+        - name: OPENSHIFT_HA_CONFIG_NAME
+          value: "ipfailover"
+        - name: OPENSHIFT_HA_VIRTUAL_IPS <1>
+          value: "1.1.1.1-2"
+        - name: OPENSHIFT_HA_VIP_GROUPS <2>
+          value: "10"
+        - name: OPENSHIFT_HA_NETWORK_INTERFACE <3>
+          value: "ens3" #The host interface to assign the VIPs
+        - name: OPENSHIFT_HA_MONITOR_PORT <4>
+          value: "30060"
+        - name: OPENSHIFT_HA_VRRP_ID_OFFSET <5>
+          value: "0"
+        - name: OPENSHIFT_HA_REPLICA_COUNT <6>
+          value: "2" #Must match the number of replicas in the deployment
+        - name: OPENSHIFT_HA_USE_UNICAST
+          value: "false"
+        #- name: OPENSHIFT_HA_UNICAST_PEERS
+          #value: "10.0.148.40,10.0.160.234,10.0.199.110"
+        - name: OPENSHIFT_HA_IPTABLES_CHAIN <7>
+          value: "INPUT"
+        #- name: OPENSHIFT_HA_NOTIFY_SCRIPT <8>
+        #  value: /etc/keepalive/mynotifyscript.sh
+        - name: OPENSHIFT_HA_CHECK_SCRIPT <9>
+          value: "/etc/keepalive/mycheckscript.sh"
+        - name: OPENSHIFT_HA_PREEMPTION <10>
+          value: "preempt_delay 300"
+        - name: OPENSHIFT_HA_CHECK_INTERVAL <11>
+          value: "2"
+        livenessProbe:
+          initialDelaySeconds: 10
+          exec:
+            command:
+            - pgrep
+            - keepalived
+      volumes:
+      - name: lib-modules
+        hostPath:
+          path: /lib/modules
+      - name: host-slash
+        hostPath:
+          path: /
+      - name: etc-sysconfig
+        hostPath:
+          path: /etc/sysconfig
+      # config-volume contains the check script
+      # created with `oc create configmap keepalived-checkscript --from-file=mycheckscript.sh`
+      - configMap:
+          defaultMode: 0755
+          name: keepalived-checkscript
+        name: config-volume
+      imagePullSecrets:
+        - name: openshift-pull-secret
```

Comment:
Should add one step of creating the `openshift-pull-secret` before creating the deployment, otherwise will get error `Failed to pull image "quay.io/openshift/ose-keepalived-ipfailover": rpc error: code = Unknown desc = Error reading manifest latest in quay.io/openshift/ose-keepalived-ipfailover: unauthorized: access to the requested resource is not authorized`